option java_package = "com.xiaoji.net";
package sails;

option cc_generic_services=true;
option java_generic_services=true;

// 排行
message RanklistRequest {
  enum Type {
    DAY = 0;
    WEEK = 1;
    MONTH = 2;
  }
  required Type type = 1;
  required int32 top = 2;
}

message RanklistResponse {
  enum ResultCode {
    SUCCESS = 0;
    ERR = 1;
  }
  message RankItem {
    required int32 rank = 1;
    required string accountId = 2;
    required int32 score = 3;
  }
  required ResultCode code = 1;
  repeated RankItem ranklist = 2;
}

// 得到用户的信息
message RankScoreRequest {
  enum Type {
    DAY = 0;
    WEEK = 1;
    MONTH = 2;
  }
  required Type type = 1;
  required string accountId = 2;
}

message RankScoreResponse {
  enum ResultCode {
    SUCCESS = 0;
    ERR = 1;
  }
  required ResultCode code = 1;
  required int32 rank = 2;
  required int32 score = 3;
}

// 增加对战结果
message RankAddFightResultRequest {
  enum Result {
    WIN = 0;
    FAILED = 1;
    ESCAPE = 2;
  }
  required Result result = 1;
  required string accountId = 2;
  required int32 gameId = 3;
  required int32 roomId = 4;
  required int32 roomtype = 5;
  required string overtime = 6;
}

message RankAddFightResultResponse {
  enum ResultCode {
    SUCCESS = 0;
    ERR = 1;
  }
  required ResultCode code = 1;
}

service RankService {
  rpc GetRanklist(RanklistRequest) returns(RanklistResponse);
  rpc GetUserScore(RankScoreRequest) returns (RankScoreResponse);
  rpc AddFightResult(RankAddFightResultRequest) returns (RankAddFightResultResponse);
}
